sklearn/ensemble/voting_classifier.py

```diff
@@ -0,0 +1,206 @@
+    >>> y = np.array([1, 1, 1, 2, 2, 2])
+    >>> eclf1 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='hard')
+    >>> eclf1 = eclf1.fit(X, y)
+    >>> print(eclf1.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf2 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft')
+    >>> eclf2 = eclf2.fit(X, y)
+    >>> print(eclf2.predict(X))
+    [1 1 1 2 2 2]
+    >>> eclf3 = VotingClassifier(estimators=[clf1, clf2, clf3], voting='soft', weights=[2,1,1])
+    >>> eclf3 = eclf3.fit(X, y)
+    >>> print(eclf3.predict(X))
+    [1 1 1 2 2 2]
+    >>>
+      
+    """
+    def __init__(self, estimators, voting='hard', weights=None):
+        self.estimators = estimators
+        self.voting = voting
+        self.weights = weights
+        self.le = LabelEncoder()
+
+    
+    def fit(self, X, y):
+        """ 
+        Fits the estimators.
+        
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+            
+        y : array-like, shape = [n_samples]
+            Target values.
+      
+        Returns
+        -------
+        self : object
+
+        """
+        
+        if self.voting not in ('soft', 'hard'):
+            raise ValueError("voting must be 'soft' or 'hard'; got (voting=%r)"
+                             % voting)
+                                                  
+        if self.weights and len(self.weights) != len(self.estimators):
+            raise ValueError('Number of classifiers and weights must be equal'\
+                         '; got %d weights, %d estimators' 
+                         % (len(self.weights), len(self.estimators)))      
+        
+        for clf in self.estimators:
+            clf.fit(X, y)
+        
+        self.le.fit(y)
+        self.classes_ = np.unique(y)
+                
+        return self
+            
+    
+    def predict(self, X):
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        maj : array-like, shape = [n_cl_lab]
+            Predicted class labels by majority rule.
+        
+        """
+        if self.voting == 'soft':
+            avg = self.predict_proba(X)
+            maj = self.le.inverse_transform(np.argmax(avg, axis=1))
+        
+        else: # 'hard' voting
+            cl_lab = self._predict(X)
+            
+            # arange class labels as [n_cl_lab, n_classifier_predictions]
+            cl_lab = np.asarray([cl_lab[:,c] 
+                        for c in range(cl_lab.shape[1])])
+            
+            # dupl. class labels if weights are provided for argmax maj. rule
+            if self.weights:
```

Comment:
can't you just use the ``weights`` parameter of ``bincount``? When was that introduced?